### PR TITLE
#184 - User authentication

### LINKED ISSUE #183: Laravel localization keys don't work with Vue implementation
Right now Laravel keys aren't translated.

Laravel localization returns some keys like `pagination.next'. We use PascalCase for our localization implemented in Vue.
It'll be nice if we adjusted these to use common format.

### LINKED ISSUE #184: User authentication
The application should have options to:
- create a new user account
- login in to an existing account
- send a password reset link

For this purpose, appropriate controllers with routes should be created and connected to existing views.

### PR DESCRIPTION
**Pull request by sbanasiak**
The site now has options for login, registration, password reset and email verification. The existing views have been adjusted for authentication functionality and new views have been created - for email verification and password reset.

image

When logging in, the user can remember his session and after logging in, user can log out by clicking a button in the navigation bar. 

image

Login and email address confirmation are now required to submit a new company - there is automatic redirect to the login/email confirmation page.

image

Routes needed for authentication have been created and a helper package [tighten/ziggy](https://github.com/tighten/ziggy) which provides a JavaScript `route()` helper function that works like Laravel's has been installed.

Closes #184 

Pull request also closes #183 - added Laravel error translations and rewritten the localization keys to snake_case.

### CODE REVIEW

File: backend/app/Http/Controllers/Auth/PasswordResetLinkController.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    /**
+     * @throws ValidationException
+     */
+    public function store(Request $request): RedirectResponse
+    {
+        $request->validate([
+            "email" => "required|email",
+        ]);
```
Comment (kamilpiech97) [inline backend/app/Http/Controllers/Auth/PasswordResetLinkController.php:31]: The same as above.

File: backend/app/Http/Controllers/Auth/RegisteredUserController.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+     * @throws ValidationException
+     */
+    public function store(Request $request): RedirectResponse
+    {
+        $request->validate([
+            "first_name" => "required|string|max:255",
+            "last_name" => "required|string|max:255",
+            "email" => "required|string|email|max:255|unique:users",
+            "password" => ["required", "confirmed", Rules\Password::defaults()],
+        ]);
```
Comment (kamilpiech97) [inline backend/app/Http/Controllers/Auth/RegisteredUserController.php:38]: The same as above.

File: frontend/js/Pages/Error.vue [hunk 3]
```diff
@@ -37,6 +37,8 @@
 </template>
 
 <script setup>
+import route from "ziggy";
+
 const props = defineProps({
   status: Number,
 });
```
Comment (kamilpiech97) [inline frontend/js/Pages/Error.vue:44]: Move setup section above of the template element - setup section is above of the template in components.

File: frontend/js/Pages/Auth/VerifyEmail.vue [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+const form = useForm();
+
+const submit = () => {
+  form.post(route("verification.send"));
+};
+
+const verificationLinkSent = computed(
+  () => props.status === "verification-link-sent"
+);
+</script>
```
Comment (kamilpiech97) [inline frontend/js/Pages/Auth/VerifyEmail.vue:66]: Move to the top of file.

**Review by kamilpiech97 — CHANGES_REQUESTED** [review]

**Review by EwelinaLasowy — COMMENTED** [review]
By default. the site is in English - maybe we should think of determining the current locale?

File: frontend/js/Pages/Auth/ResetPassword.vue [hunk 5]
```diff
@@ -0,0 +1,111 @@
+<script setup>
```
[the remark this answers is not in the document]
Comment (sbanasiak) [reply frontend/js/Pages/Auth/ResetPassword.vue:1]: Fixes to the translation of the site and other elements will be included in a separate localization pull request.

**Review by kamilpiech97 — APPROVED** [review]

**Review by EwelinaLasowy — APPROVED** [review]

**Pull request merged**